src/app/shared/Compose.svelte

```diff
@@ -1,26 +1,105 @@
+<style>
+  :global(.lazy-image) {
+    animation: blurPulse 1.5s infinite;
+  }
+
+  :global(a[data-type="tag"]) {
+    text-decoration: underline;
+  }
+
+  @keyframes blurPulse {
+    0% {
+      filter: blur(0px);
+    }
+    50% {
+      filter: blur(15px);
+    }
+    100% {
+      filter: blur(0px);
+    }
+  }
+</style>
+
 <script lang="ts">
   import {nip19} from "nostr-tools"
   import {throttle} from "throttle-debounce"
-  import {createEventDispatcher} from "svelte"
-  import {whereEq} from "ramda"
+  import {createEventDispatcher, onMount} from "svelte"
   import {ctx, last, partition} from "@welshman/lib"
-  import {displayProfileByPubkey, profileSearch} from "@welshman/app"
+  import {profileSearch} from "@welshman/app"
   import PersonBadge from "src/app/shared/PersonBadge.svelte"
-  import ContentEditable from "src/partials/ContentEditable.svelte"
   import Suggestions from "src/partials/Suggestions.svelte"
-  import {userFollows, createPeopleLoader} from "src/engine"
+  import {userFollows, createPeopleLoader, getSetting} from "src/engine"
+  import type {Readable} from "svelte/store"
+  import {createEditor, Editor, SvelteNodeViewRenderer} from "svelte-tiptap"
+  import StarterKit from "@tiptap/starter-kit"
+  import {NostrExtension} from "nostr-editor"
+  import {signer} from "@welshman/app"
+  import PersonLink from "src/app/shared/PersonLink.svelte"
 
   export let onSubmit
   export let autofocus = false
   export let placeholder = null
-
-  let contenteditable, suggestions
+  export let hostLimit = 1
+
+  let editor: Readable<Editor>
+
+  let suggestions
+
+  let element: HTMLDivElement
+
+  $: textBeforeCursor = $editor?.state?.selection?.$anchor?.nodeBefore?.text || ""
+
+  onMount(() => {
+    const urls = getSetting("nip96_urls").slice(0, hostLimit)
+    editor = createEditor({
+      autofocus,
+
+      element: element,
+      editorProps: {
+        attributes: {
+          placeholder,
+        },
+      },
+
+      extensions: [
+        StarterKit,
```

Comment:
StarterKit has lots of stuff we don't want to support. See flotilla's set up for something much closer to reality.